RELEASE.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+# Release process automation script. 
+# Used to create branch/tag, update VERSION files and and trigger release by force pushing changes to the release branch 
+
+# set to 1 to actually trigger changes in the release branch
+TRIGGER_RELEASE=0 
+
+#REPO=git@github.com:eclipse/che-machine-exec
+#VERSION=7.7.0
+
+while [[ "$#" -gt 0 ]]; do
+  case $1 in
+    '-t'|'--trigger-release') TRIGGER_RELEASE=1; shift 0;;
+    '-r'|'--repo') REPO="$2"; shift 1;;
+    '-v'|'--version') VERSION="$2"; shift 1;;
+  esac
+  shift 1
+done
+
+usage ()
+{
+  echo "Usage: $0 --repo [GIT REPO TO EDIT] --version [VERSION TO RELEASE] [--trigger-release]"
+  echo "Example: $0 --repo git@github.com:eclipse/che-machine-exec --version 7.7.0 --trigger-release"; echo
+}
+
+if [[ ! ${VERSION} ]] || [[ ! ${REPO} ]]; then
+  usage
+  exit 1
+fi
+
+# derive branch from version
+BRANCH=${VERSION%.*}.x
+
+# work in tmp dir
+TMP=$(mktemp -d); mkdir -p "$TMP"; cd "$TMP" || exit 1
+
+# get sources from master branch
+echo "Check out ${REPO} to ${TMP}/${REPO##*/}"
+git clone "${REPO}" -q
+cd "${REPO##*/}" || exit 1
+git fetch origin master:master
+git checkout master
+
+# create new branch off master + push to origin
+git branch "${BRANCH}" || git checkout "${BRANCH}" && git pull origin "${BRANCH}"
+git push origin "${BRANCH}"
+git fetch origin "${BRANCH}:${BRANCH}"
+git checkout "${BRANCH}"
```

Comment:
This sequence isn't clear to me -- why do we need to checkout `${BRANCH}` multiple times?

Reply:
if creating the branch fails, then I figured I'd checkout and pull instead